File: app/src/main/java/com/bitflaker/lucidsourcekit/setup/ViewPagerAdapter.java
package com.bitflaker.lucidsourcekit.setup;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.lifecycle.Lifecycle;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import java.util.ArrayList;

public class ViewPagerAdapter extends FragmentStateAdapter {

    private final ArrayList<Fragment> fragmentArrayList = new ArrayList<>();
    private final ArrayList<String> fragmentTitle = new ArrayList<>();

    public ViewPagerAdapter(@NonNull FragmentManager fragmentManager, @NonNull Lifecycle lifecycle) {
        super(fragmentManager, lifecycle);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        return fragmentArrayList.get(position);
    }

    @Override
    public int getItemCount() {
        return fragmentArrayList.size();
    }

    public void addFragment(Fragment fragment, String title) {
        fragmentArrayList.add(fragment);
        fragmentTitle.add(title);
    }

    public void removeFragment(Fragment fragment, String title) {
        fragmentArrayList.remove(fragment);
        fragmentTitle.remove(title);
    }

    public Fragment getFragment(String title) {
        return fragmentArrayList.get(fragmentTitle.indexOf(title));
    }

    public int getTabIndex(String title) {
        return fragmentTitle.indexOf(title);
    }
}

File: app/src/main/java/com/bitflaker/lucidsourcekit/ViewPagerAdapter.kt
package com.bitflaker.lucidsourcekit

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.lifecycle.Lifecycle
import androidx.viewpager2.adapter.FragmentStateAdapter

class ViewPagerAdapter(fragmentManager: FragmentManager, lifecycle: Lifecycle) : FragmentStateAdapter(fragmentManager, lifecycle) {
    private val fragmentArrayList = ArrayList<Fragment>()
    private val fragmentTitle = ArrayList<String>()

    override fun createFragment(position: Int): Fragment = fragmentArrayList[position]
    override fun getItemCount(): Int = fragmentArrayList.size

    fun getFragment(title: String): Fragment = fragmentArrayList[fragmentTitle.indexOf(title)]
    fun getTabIndex(title: String): Int = fragmentTitle.indexOf(title)
    fun isCurrentPage(title: String, position: Int): Boolean = fragmentTitle.size > position && fragmentTitle[position] == title

    fun addFragment(fragment: Fragment, title: String) {
        fragmentArrayList.add(fragment)
        fragmentTitle.add(title)
    }

    fun removeFragment(fragment: Fragment, title: String) {
        fragmentArrayList.remove(fragment)
        fragmentTitle.remove(title)
    }
}